xonsh/__init__.py

```diff
@@ -26,8 +26,6 @@
         _sys.modules['xonsh.platform'] = __amalgam__
         pretty = __amalgam__
         _sys.modules['xonsh.pretty'] = __amalgam__
-        jobs = __amalgam__
```

Comment:
They are totally reliable and deterministic. The issue is that changes to the code (imports specifically) can change the dependency graph of the different modules. This is then reflected in the ordering here.